### ISSUE TITLE
$?#3f failed

### ISSUE THREAD

**Issue by kenz-gelsoft** [kenz-gelsoft/llvm-project#15]
```
 <  68> send packet: $A58,0,2f626f6f742f686f6d652f7372632f6370705f746573742f612e6f7574#72
 this = 0x0000103F3BBA1F70, dst = 0x00007F4DFAE2F8B0, dst_len = 8192, timeout = 10000000 us, connection = 0x0000103F3BA75360
 this = 0x0000103F3BA75360, timeout = 10000000 us
 0x103f3ba75360 ConnectionFileDescriptor::Read()  fd = 4, dst = 0x7f4dfae2f8b0, dst_len = 8192) => 6, error = (null)
 <   6> read packet: $OK#9a
 0x0000103F3BBA1F70 Communication::Write (src = 0x0000103F3BA7F378, src_len = %lu) connection = 18
 0x103f3ba75360 ConnectionFileDescriptor::Write (src = 0x103f3ba7f378, src_len = 18)
 0x103f3ba75360 ConnectionFileDescriptor::Write(fd = 4, src = 0x103f3ba7f378, src_len = 18) => 18 (error = (null))
 <  18> send packet: $qLaunchSuccess#a5
 this = 0x0000103F3BBA1F70, dst = 0x00007F4DFAE2F960, dst_len = 8192, timeout = 10000000 us, connection = 0x0000103F3BA75360
 this = 0x0000103F3BA75360, timeout = 10000000 us
 0x103f3ba75360 ConnectionFileDescriptor::Read()  fd = 4, dst = 0x7f4dfae2f960, dst_len = 8192) => 6, error = (null)
 <   6> read packet: $OK#9a
 0x0000103F3BBA1F70 Communication::Write (src = 0x0000103F3BA7F378, src_len = %lu) connection = 16
 0x103f3ba75360 ConnectionFileDescriptor::Write (src = 0x103f3ba7f378, src_len = 16)
 0x103f3ba75360 ConnectionFileDescriptor::Write(fd = 4, src = 0x103f3ba7f378, src_len = 16) => 16 (error = (null))
 <  16> send packet: $qProcessInfo#dc
 this = 0x0000103F3BBA1F70, dst = 0x00007F4DFAE2F5B0, dst_len = 8192, timeout = 10000000 us, connection = 0x0000103F3BA75360
 this = 0x0000103F3BA75360, timeout = 10000000 us
 0x103f3ba75360 ConnectionFileDescriptor::Read()  fd = 4, dst = 0x7f4dfae2f5b0, dst_len = 8192) => 168, error = (null)
 < 168> read packet: $pid:2abe;parent-pid:2abb;real-uid:0;real-gid:0;effective-uid:0;effective-gid:0;triple:7838365f36342d756e6b6e6f776e2d6861696b75;ostype:haiku;endian:little;ptrsize:8;#0b
 0x0000103F3BBA1F70 Communication::Write (src = 0x0000103F3BAF95F8, src_len = %lu) connection = 5
 0x103f3ba75360 ConnectionFileDescriptor::Write (src = 0x103f3baf95f8, src_len = 5)
 0x103f3ba75360 ConnectionFileDescriptor::Write(fd = 4, src = 0x103f3baf95f8, src_len = 5) => 5 (error = (null))
 <   5> send packet: $?#3f
 this = 0x0000103F3BBA1F70, dst = 0x00007F4DFAE2F9C0, dst_len = 8192, timeout = 5000000 us, connection = 0x0000103F3BA75360
 this = 0x0000103F3BA75360, timeout = 5000000 us
 0x103f3ba75360 ConnectionFileDescriptor::Read()  fd = 4, dst = 0x7f4dfae2f9c0, dst_len = 8192) => 0, error = (null)
 0x0000103F3BBA1F70 Communication::Disconnect ()
 MonitorChildProcessThreadFunction ::waitpid (pid = 10939, &status, options = 0) => pid = 10939, status = 0x00000001 (EXITED), signal = 0, exit_state = 1
 0x103f3ba75360 ConnectionFileDescriptor::Disconnect ()
 ProcessGDBRemote::MonitorDebugserverProcess(process_wp, pid=10939, signo=0 (0x0), exit_status=1)
 ProcessGDBRemote::MonitorDebugserverProcess(process = 0x103f3bba16a0)
 timeout = 10000000 us, event_sp)...
 this = 0x0000103F3BA427C0, timeout = 10000000 us for lldb.process.internal_state_listener
 Process::SetExitStatus (status=-1 (0xffffffff), description="debugserver died with an exit status of 0x00000001")
 Process::SetPrivateState (exited)
 Process::SetPrivateState (exited) stop_id = 1
 0x103f3baee180 Broadcaster("lldb.process.internal_state_broadcaster")::BroadcastEvent (event_sp = {0x103f3bbc6430 Event: broadcaster = 0x103f3bba1760 (lldb.process.internal_state_broadcaster), type = 0x00000001, data = { process = 0x103f3bba16a0 (pid = 10942), state = exited}}, unique =0) hijack = (nil)
 0x103f3ba427c0 Listener('lldb.process.internal_state_listener')::AddEvent (event_sp = {0x103f3bbc6430})
 0x0000103F3BBA1F70 Communication::Disconnect ()
 0x103f3ba427c0 'lldb.process.internal_state_listener' Listener::FindNextEventInternal(broadcaster=0x103f3bba1760, broadcaster_names=(nil)[0], event_type_mask=0x00000003, remove=1) event 0x103f3bbc6430
 0x103f3ba75360 ConnectionFileDescriptor::Disconnect ()
 timeout = 10000000 us, event_sp) => exited
 0x103f3ba75360 ConnectionFileDescriptor::Disconnect(): Nothing to disconnect
 0x103f3bb3f9d0 Listener::Listener('Communication::SyncronizeWithReadThread')
 MonitorChildProcessThreadFunction (arg = 0x103f3bb8e660) thread exiting because pid received exit signal...
 0x103f3bb3f9d0 Listener::StartListeningForEvents (broadcaster = 0x103f3bba1b28, mask = 0x00000020) acquired_mask = 0x00000020 for Communication::SyncronizeWithReadThread
 0x103f3bb3f9d0 Listener::Clear('Communication::SyncronizeWithReadThread')
 MonitorChildProcessThreadFunction (arg = 0x103f3bb8e660) thread exiting...
 0x103f3bb3f9d0 Listener::~Listener('Communication::SyncronizeWithReadThread')
 0x0000103F3BBA1B28 Communication::Disconnect ()
 Process::ShouldBroadcastEvent (0x103f3bbc6430) => new state: exited, last broadcast state: exited - YES
 Process::HandlePrivateEvent (pid = 10942) broadcasting new state exited (old state launching) to hijacked
 0x103f3baee050 Broadcaster("lldb.process")::BroadcastEvent (event_sp = {0x103f3bbc6430 Event: broadcaster = 0x103f3bba16d8 (lldb.process), type = 0x00000001 (state-changed), data = { process = 0x103f3bba16a0 (pid = 10942), state = exited}}, unique =0) hijack = 0x103f3ba42980
 0x103f3ba42980 Listener('lldb.PlatformLinux.DebugProcess.hijack')::AddEvent (event_sp = {0x103f3bbc6430})
 timeout = <infinite>
 timeout = <infinite>, event_sp)...
 this = 0x0000103F3BA42980, timeout = <infinite> for lldb.PlatformLinux.DebugProcess.hijack
 0x103f3ba42980 'lldb.PlatformLinux.DebugProcess.hijack' Listener::FindNextEventInternal(broadcaster=0x103f3bba16d8, broadcaster_names=(nil)[0], event_type_mask=0x00000003, remove=1) event 0x103f3bbc6430
 Process::SetPublicState (state = exited, restarted = 0)
 timeout = <infinite>, event_sp) => exited
 pid 10942 state 10942
 0x103f3ba754a0 ConnectionFileDescriptor::ConnectionFileDescriptor (fd = 3, owns_fd = 1)
 0x103f3ba754a0 ConnectionFileDescriptor::CloseCommandPipe()
 0x103f3ba754a0 ConnectionFileDescriptor::OpenCommandPipe() - success readfd=4 writefd=5
 0x0000103F3BBA1B28 Communication::Disconnect ()
 0x0000103F3BBA1B28 Communication::StartReadThread ()
 thread created
 hooked up STDIO pty to process
 0x103f3bba1b28 Communication::ReadThread () thread starting...
 0x0000103F3BAEE050 Broadcaster("lldb.process")::RestoreBroadcaster (about to pop listener("lldb.PlatformLinux.DebugProcess.hijack")=0x0000103F3BA42980)
 this = 0x0000103F3BA754A0, timeout = 5000000 us
 HandleCommand, command did not succeed
error: process exited with status -1 (debugserver died with an exit status of 0x00000001)
'r' and 'run' are aliases that default to launching through a shell.
Try launching without going through a shell by using 'process launch'.
(lldb)  error: timed out, status = timed out
 this = 0x0000103F3BA754A0, timeout = 5000000 us
 error: timed out, status = timed out
 this = 0x0000103F3BA754A0, timeout = 5000000 us
 error: timed out, status = timed out
 this = 0x0000103F3BA754A0, timeout = 5000000 us
 error: timed out, status = timed out
```

**Comment by kenz-gelsoft** [comment]
```
    eServerPacketType_stop_reason, // '?'
```

**Comment by kenz-gelsoft** [comment]
Fixed by 11b115d4aac834c75ee5add7f3120f237a4b9157
